회원가입 & 로그인 기능 구현

twnth opened the pull request.

twnth committed:
refactor: http 통신 위한 클래스들 추가.

twnth committed:
refactor: AccountApi 로 수정, repo 모듈 scope 수정

twnth committed:
refactor: join, login으로 향하는 navigate 생성(이렇게 하는게 맞나...?)

twnth committed:
refactor: api reponse 구조 수정

twnth committed:
refactor: LoginScreen 구현, login api 연동

NackUn commented on app/src/main/java/com/dogandpigs/fitnote/core/Constants.kt:
```diff
@@ -3,6 +3,10 @@ package com.dogandpigs.fitnote.core
 
+}
+
+object ResCode {
```
ResponseCode로 줄이지 않고 사용하는 것도 좋을 것 같습니다.

NackUn commented on app/src/main/java/com/dogandpigs/fitnote/core/TokenManager.kt:
```diff
@@ -0,0 +1,13 @@
+    fun setAccessToken(token: String) {
+        accessToken = token
+    }
+}
```
토큰 관리는 DataStore(SharedPreference)로 관리하면 어떨까요?

NackUn commented on app/src/main/java/com/dogandpigs/fitnote/data/repository/AccountRepository.kt:
```diff
@@ -0,0 +1,34 @@
+class AccountRepository @Inject constructor(
+    private val api: AccountApi
+) {
+    suspend fun join(data: JoinRequestDTO): Flow<UserDTO?> {
```
Flow말고 UserDTO로 받아도 괜찮지않을까요?

NackUn commented on app/src/main/java/com/dogandpigs/fitnote/data/source/remote/model/JoinRequestDTO.kt:
```diff
@@ -0,0 +1,17 @@
+
+import com.google.gson.annotations.SerializedName
+
+data class JoinRequestDTO(
```
JoinRequestDTO -> JoinRequest는 어떨까요?

NackUn reviewed (commented):
고생하셨습니다!
UI 쪽 제외하고 리뷰 남겼습니다

twnth committed:
feat: Token용 SharePreference 구현

twnth committed:
refactor: PR  피드백 반영

1. ResCode -> ResponseCode 수정
2. AccountRepository의 api request 반환값 Flow
-> DTO로 수정
3. JoinRequestDTO -> JoinRequest로 수정

twnth committed:
refactor: NavigationBar visible 처리 추가

Splash 단계에서는 안보이게 처리

twnth committed:
refactor: login 후 TokenManager에 token 세팅 추가

NackUn reviewed (approved):
고생하셨습니다!
